### PR TITLE
Update fill and get gaps methods

### PR DESCRIPTION
**Pull request by saelsa**

### CODE REVIEW

File: tests/test_timeseriesx.py [hunk 1]
```diff
@@ -817,6 +817,32 @@ def test_timestamp_series_fill_gaps():
...
+        dt.datetime(2020, 3, 1, 17, 0, 0),
+    ]
+    values = [1, 1, np.nan]
+    ts = TimestampSeries.create_from_lists(timestamps, values, freq=pd.offsets.Hour())
+    ts = ts.fill_gaps(value=2)
+    assert len(ts) == 3
+    assert ts.values == [1, 1, 2]
+
+
+def test_timestamp_series_fill_gaps_with_unit(default_timestamp_series):
```
Comment (carlculator) [inline tests/test_timeseriesx.py:878]: this test actually covers two test cases. please split into two. 
* __first__: `test_timestamp_series_fill_gaps_with_start_end`:
    * do not focus on units here  
    * please assert also on the timestamps, not only the values
    * you may also provide a value for start
* __second__: `test_timestamp_series_fill_gaps_with_unit`:
    * test filling a dimensionless number into a unit based series
    * assert on integrity of the `_series`-dtype

**Comment by carlculator** [comment]
Can you add a test where time zones of `start` and `end` in `get_gaps` do not match the `self.time_zone` of the `TimestampSeries`. Please use the `_get_time_zone`-method in the `FrequencyMixin` for this. How do we handle time zone naive `start`s and `end`s in this case? I'd advocate interpreting them as matching the `TimestampSeries`'s time zone.

File: tests/test_timeseriesx.py (hunk 1 above)
Comment (saelsa) [reply tests/test_timeseriesx.py:878]: The first part is already implemented in the `test_timestamp_series_fill_gaps`, no? Or would you prefer additional checks? For the second one I added the dtype-check to the test

**Comment by saelsa** [comment]
I have added checks and additional tests for not matching start and end timezones. I agree with interpreting time zone naive parameters in the time series timezone

File: timeseriesx/mixins/frequency.py [hunk 2]
```diff
@@ -23,43 +23,77 @@ def __init__(self, *args, **kwargs):
...
+        if (not start or not end) and self._series.empty:
+            raise ValueError('cannot fill the gaps for empty series '
+                             'without parameters providing start and end.')
+
+        start = start or self._series.index[0].to_pydatetime()
+        end = end or self._series.index[-1].to_pydatetime()
+        time_zone = self._get_time_zone()
+
+        if start.tzinfo and start.tzinfo != time_zone:
+            raise ValueError('time zone of parameter start does not match '
```
Comment (carlculator) [inline timeseriesx/mixins/frequency.py:52]: I would go for a user friendly approach here. as long as the time zone can be converted, it should be fine. there should be no error. I guess pandas does this automatically

File: tests/test_timeseriesx.py (hunk 1 above)
Comment (carlculator) [reply tests/test_timeseriesx.py:878]: There should be a test for each public method. Indeed, the `get_gaps` relies on `fill_gaps`. But i don't want to have tests based on assumptions on internals of the implementation of public methods, rather on their signature and expected behaviour.

Comment (carlculator) [reply tests/test_timeseriesx.py:878]: Sorry, I oversaw the existing tests.

File: timeseriesx/mixins/frequency.py (hunk 2 above)
Comment (carlculator) [reply timeseriesx/mixins/frequency.py:52]: OK. we go for the error approach

**Review by carlculator — APPROVED** [review]
ok. thanks a lot for your contribution

**Pull request merged**